src/js/components/TextInput/TextInput.js

```diff
@@ -106,6 +106,12 @@ const TextInput = forwardRef(
     const { format } = useContext(MessageContext);
     const announce = useContext(AnnounceContext);
     const formContext = useContext(FormContext);
+    if (formContext.noForm !== undefined && !formContext.noForm && !name) {
```

Comment:
Over here, `undefined` is added because there is no name prop added for TextInput in DataTableColumns component